fendermint/storage/src/lib.rs

```diff
@@ -156,4 +166,18 @@ where
     pub fn delete(&self, kv: &mut impl KVWrite<S>, k: &K) -> KVResult<()> {
         kv.delete(&self.ns, k)
     }
+
+    pub fn iterate<'a, 'b>(
+        &'a self,
+        kv: &'b impl KVRead<S>,
+    ) -> impl Iterator<Item = KVResult<(K, V)>> + 'b
+    where
+        S::Repr: Ord + 'static,
+        S: Decode<K>,
+        K: 'static,
+        V: 'static,
+        'a: 'b,
```

Comment:
I recall seeing this before, but its one of these lifetimes I never remember how is written

Reply:
This is also from the compiler. I didn't think adding this method on `KVCollection` would involve more constraints. It complained that the `impl Interator` captures the lifetimes of both reference inputs, so it needs to outlive them, and this was the only way to express that they are the same, or compatible.